### DISCUSSION TITLE
aws-lmbda-plugin: add support for the ExternalId for the cross account lambda invocation.

### DISCUSSION THREAD

**Discussion by shambhand** [Kong/kong#13737] [Ideas and feature requests]
Currently the aws lambda plugin does not provide the way for external id aws config for generating temporary credential.

https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html

This would be helpful for the large organisation which is having multiple aws account and hosted the shared kong use by the multiple Team and want to access the lambda hosted in another aws account. The kong is using the kubernetes IAM role for service account. I can understand this can be done by trusting the kong irsa role by the lambda invocation. but requires to notify and update the Teams which owns the lambda invocation role and if the roles changes this would be maintenance overhead so aws credential with external id  can help in this case

**Comment by ProBrian** [comment]
Hi @shambhand, our product and engineering team are keeping improving the plugin functionality, but as I know we don't have a plan to support external ID in the plugin in short-term, and there are no similar requirement from our enterprise customers. 
Anyway, thank you for your suggestion and we may put this in our future consideration.
